sdk/storage/Azure.Storage.DataMovement/tests/PauseResumeTransferMockedTests.cs

```diff
@@ -0,0 +1,826 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Buffers;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Azure.Core;
+using Azure.Core.Pipeline;
+using Moq;
+using NUnit.Framework;
+using static Azure.Storage.DataMovement.Tests.MemoryTransferCheckpointer;
```

Comment:
is this line needed? MemoryTransferCheckpointer lives in the same namespace as these tests right?

Reply:
@jaschrep-msft Can we change `MemoryTransferCheckpointer` to public ? It's only something used for our tests

Reply:
I'm not sure if we can just change MemoryTransferCheckpointer to public. Seems like it would lead to inconsistent accessibility since the properties JobPlanOperation and JobPartPlanHeader are both internal. We would have to change both JobPlanOperation and JobPartPlanHeader to public as well

Reply:
Yeah there def are workarounds for this. The quickest way is just using 'var' instead of explicit typing. You can also play around with the accessibility levels as well.